crates/fj-core/src/validate/sketch.rs

```diff
@@ -1,12 +1,140 @@
-use crate::objects::Sketch;
+use crate::{objects::Sketch, validate_references};
 
-use super::{Validate, ValidationConfig, ValidationError};
+use super::{
+    references::{ReferenceCountError, ReferenceCounter},
+    Validate, ValidationConfig, ValidationError,
+};
 
 impl Validate for Sketch {
     fn validate_with_config(
         &self,
-        _: &ValidationConfig,
-        _: &mut Vec<ValidationError>,
+        config: &ValidationConfig,
+        errors: &mut Vec<ValidationError>,
     ) {
+        SketchValidationError::check_obect_references(self, config, errors);
+    }
+}
+
+/// [`Sketch`] validation failed
+#[derive(Clone, Debug, thiserror::Error)]
+pub enum SketchValidationError {
+    /// Object within sketch referenced by more than one other object
+    #[error("Object within sketch referenced by more than one other Object")]
+    MultipleReferences(#[from] ReferenceCountError),
+}
+
+impl SketchValidationError {
+    fn check_obect_references(
```

Comment:
```suggestion
    fn check_object_references(
```

(Also requires changing the caller, of course, so just clicking "accept" on this suggestion won't be enough.)